File: VacSlotBookingApp/src/app/services/slot.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { HospitalDetails } from '../model/hospital-details.model';
import { PatientRecord } from '../model/patient-record.model';

@Injectable({
  providedIn: 'root'
})
export class SlotService {

  constructor(private http: HttpClient) { }
  addPatientRecord(patientRecord:PatientRecord) {
    return this.http.post<number>(environment.apiBaseUrl + "Slot/AddPatientRecord",patientRecord);
  }
  getAllPatientRecords() {
    return this.http.get<PatientRecord[]>(environment.apiBaseUrl + "Slot/GetAllPatientRecords");
  } 
  getDashboardDetails(HospitalId:number,SlotId:number,SlotDate:Date) {
    return this.http.get<HospitalDetails[]>(environment.apiBaseUrl + "Slot/GetDashboardDetails/"+HospitalId+"/"+SlotId+"/"+SlotDate);
  } 
  getPatientRecordsByFilter(HospitalId:number,SlotId:number,SlotDate:Date) {
    return this.http.get<PatientRecord[]>(environment.apiBaseUrl + "Slot/GetPatientRecordsByFilter/"+HospitalId+"/"+SlotId+"/"+SlotDate);
  } 
  getAllHospitals() {
    return this.http.get<HospitalDetails[]>(environment.apiBaseUrl + "Slot/GetAllHospitals");
  } 
  getAllSlots() {
    return this.http.get<HospitalDetails[]>(environment.apiBaseUrl + "Slot/GetAllSlots");
  } 
  getSlotBookingsById(CreatedBy:number) {
    return this.http.get<PatientRecord[]>(environment.apiBaseUrl + "Slot/GetSlotBookingsById/"+CreatedBy);
  } 
}

File: VacSlotbookingAPI/Business/Implementations/SlotService.cs
using Business.Interfaces;
using Domain;
using System;
using Repository.Interfaces;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Business.Implementations
{
    public class SlotService : ISlotService
    {
        ISlotRepository _studentRepository;

        public SlotService(ISlotRepository studentRepository)
        {
            _studentRepository = studentRepository;
        }

        public async Task<int> AddPatientRecord(PatientRecord patientRecord)
        {
            return await _studentRepository.AddPatientRecord(patientRecord);
        }
        public async Task<IEnumerable<PatientRecord>> GetAllPatientRecords()
        {
            return await _studentRepository.GetAllPatientRecords();
        }
        public async Task<IEnumerable<PatientRecord>> GetSlotBookings(int CreatedBy)
        {
            return await _studentRepository.GetSlotBookings(CreatedBy);
        }
        public async Task<IEnumerable<HospitalDetails>> GetDashboardDetails(int HospitalId, int SlotId, DateTime SlotDate)
        {
            return await _studentRepository.GetDashboardDetails(HospitalId,SlotId,SlotDate);
        }
        public async Task<IEnumerable<PatientRecord>> GetPatientRecordsByFilter(int HospitalId, int SlotId, DateTime SlotDate)
        {
            return await _studentRepository.GetPatientRecordsByFilter(HospitalId, SlotId, SlotDate);
        }
        public async Task<IEnumerable<HospitalDetails>> GetAllHospitals()
        {
            return await _studentRepository.GetAllHospitals();
        }
        public async Task<IEnumerable<HospitalDetails>> GetAllSlots()
        {
            return await _studentRepository.GetAllSlots();
        }
    }
}
